Foundation/FileManager+Win32.swift

```diff
@@ -648,7 +644,95 @@ extension FileManager {
     }
 
     internal func _contentsEqual(atPath path1: String, andPath path2: String) -> Bool {
-        NSUnimplemented()
+        guard let path1Handle = path1.withCString(encodedAs: UTF16.self, {
+            CreateFileW($0, GENERIC_READ, DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE), nil,
+                        DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS), nil)
+        }) else {
+            return false
+        }
+        defer { CloseHandle(path1Handle) }
+
+        guard let path2Handle = path2.withCString(encodedAs: UTF16.self, {
+            CreateFileW($0, GENERIC_READ, DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE), nil,
+                        DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS), nil)
+        }) else {
+            return false
+        }
+        defer { CloseHandle(path2Handle) }
+
+        guard path1Handle != INVALID_HANDLE_VALUE, path2Handle != INVALID_HANDLE_VALUE else {
```

Comment:
Should these not be before the `defer`?

Reply:
I swore I read somewhere it was safe to call CloseHandle on an `INVALID_HANDLE_VALUE` but the docs definitely say it's not...